spec/api-system-preferences-spec.js

```diff
@@ -35,6 +35,35 @@ describe('systemPreferences module', () => {
     })
   })
 
+  describe('systemPreferences.registerDefaults(defaults)', () => {
+    before(function () {
+      if (process.platform !== 'darwin') this.skip()
+    })
+
+    it('registers defaults', () => {
+      const userDefaults = {
+        'one': 'onee',
+        'two': 'twoo',
+        'three': 'threee',
+        'four': 'fourr',
+        'five': 'fivee'
+      }
+
+      systemPreferences.registerDefaults(userDefaults)
+
+      const val = systemPreferences.getUserDefault('two', 'string')
+      assert.equal(val, 'twoo')
+    })
+
+    it('throws when bad defaults are passed', () => {
+      const userDefaults = 1
```

Comment:
Should we test any other invalid values? Like an empty object, `null`, `undefined`, `Number.NaN`, object with valid keys but with invalid/empty values? What result should be expected when they are used?

Reply:
it crashes on `null` & `undefined`, i'll catch for that